src/DynamoCoreWpf/ViewModels/Core/OutPortViewModel.cs

```diff
@@ -309,16 +309,7 @@ private void OnMouseLeftButtonDownOnContext(object parameter)
 
         protected override void RefreshPortColors()
         {
-            var isCachedValueNull = node.NodeModel.CachedValue == null || node.NodeModel.CachedValue.Data == null;
-
-            if (node.NodeModel.CachedValue != null  && node.NodeModel.CachedValue.IsFunction)
-            {
-                PortDefaultValueMarkerVisible = true;
-            }
-            else if (isCachedValueNull)
-            {
-                PortDefaultValueMarkerVisible = false;
-            }
+            PortDefaultValueMarkerVisible = node.NodeModel.CachedValue != null && node.NodeModel.CachedValue.IsFunction;
```

Comment:
Do we still need to check `node.NodeModel.CachedValue.Data == null`?

Reply:
when node.NodeModel.CachedValue.Data is null, IsFunction will be false and the port marker will not be visible.